src/pal/src/configure.cmake

```diff
@@ -33,7 +33,9 @@ check_include_files(sys/lwp.h HAVE_SYS_LWP_H)
 check_include_files(lwp.h HAVE_LWP_H)
 check_include_files(libunwind.h HAVE_LIBUNWIND_H)
 check_include_files(runetype.h HAVE_RUNETYPE_H)
+set(CMAKE_REQUIRED_FLAGS "-ldl")
```

Comment:
Thanks @janvorli.  I tried to use CMAKE_REQUIRED_LIBRARIES instead, but it seems that check_include_files does not look at it.  I've confirmed this in the documentation: https://cmake.org/cmake/help/v3.0/module/CheckIncludeFiles.html.

I have modified the change to use the conditions you specified above.